src/OrleansRuntime/Streams/PersistentStream/PersistentStreamPullingManager.cs

```diff
@@ -89,8 +89,7 @@ internal class PersistentStreamPullingManager : SystemTarget, IPersistentStreamP
             // Remove cast once we cleanup
             queueAdapter = qAdapter.Value;
             await this.queueBalancer.Initialize(this.streamProviderName, this.adapterFactory.GetStreamQueueMapper(), config.SiloMaturityPeriod, this.providerConfig);
-            var meAsQueueBalanceListener = this.AsReference<IStreamQueueBalanceListener>();
-            queueBalancer.SubscribeToQueueDistributionChangeEvents(meAsQueueBalanceListener);
+            queueBalancer.SubscribeToQueueDistributionChangeEvents(this);
```

Comment:
How do you feel this general problem should be addressed?
* OS Thread notifies some Orleans component of some event
* That component propagates that notification to subscribers

Should the component marshal the notification onto an addressable context?
Alternatively, should the subscriber be responsible for ensuring that they execute on the correct context?

I am in favor of the latter approach, since the first is difficult without some publicly exposed mechanism for doing so (eg, SystemTarget - which the SF code does not currently have access to) and also because it's the subscriber which knows what its requirements are - not the publisher.